modules/apps/portal-search/portal-search-web/src/main/resources/META-INF/resources/js/components/SearchBar.js

```diff
@@ -171,8 +173,18 @@ export default function SearchBar({
 		event.stopPropagation();
 
 		if (!!inputValue.trim().length || emptySearchEnabled) {
+			const keyword = inputValue.trim();
+			const oldKeyword = localStorage.getItem('keyword');
 			const queryString = _updateQueryString(document.location.search);
 
+			if (
+				(keyword === '' || keyword !== oldKeyword) &&
+				!retainFacetSelectionsAcrossSearches
+			) {
+				FacetUtil.clearAllSelections(queryString);
+			}
+
+			localStorage.setItem('keyword', keyword);
```

Comment:
Is the first search's keyword available in the search response? I think it would be better to compare that to the new keyword.

Also, does this work properly when the user clicks the back browser button?

Reply:
If we need to name the first search's keyword, maybe "previousKeyword" would be good.